### PR TITLE
Improve setup docs

### LINKED ISSUE #420: Improve developer setup instructions
### Definition of Done / Acceptance Criteria

- [x] Clear instructions on how to install the various Python modules (e.g. note that the CLI needs to be installed in editable mode`)
- [x] Add brief overview of running module-level tests and integration tests to developer docs

### Testing

_No response_

### Documentation

- [x] Update top-level README with user installation instructions
- [x] Update `docs/setup/developer.md` with developer installation instructions
- [x] Remove outdated docs on available CLI commands

### Dependencies

- [ ] Related to #247

### Details and Comments

_No response_

### PR DESCRIPTION
**Pull request by p-j-smith**
Fixes #420

- provide commands to install all Python modules (both as a user and a developer)
- add short description of how to run module and integration tests to developer docs
- use `python -m pip install` rather than `pip install` (it's the way [recommended by PyPA](https://pip.pypa.io/en/stable/user_guide/#running-pip) to ensure you're using the correct pip)
- add quotes around the extras to install with pip (required by zsh and some other shells)
- remove docs on currently available commands with the PIXL CLI as the docs are out of date. Instead, provide command users can type to find out currently available commands

### CODE REVIEW

File: docs/setup/developer.md [hunk 1]
```diff
@@ -1,21 +1,54 @@
 # Developer setup
 
+## Installation
+
+You can install all PIXL Python modules by running the following commands from the `PIXL/` directory:
+
+```shell
+python -m pip install -e "pixl_core/[dev]"
+python -m pip install -e "pytest-pixl/[dev,test]"
+python -m pip install -e "pixl_core/[test]"
```
Comment (p-j-smith) [inline docs/setup/developer.md:10]: this seems convoluted but is necessary as `pixl_core[test]` depends on `pytest_pixl`, and `pytest_pixl[test]` depends on `pixl_core`

File: README.md [hunk 2]
```diff
@@ -15,37 +15,28 @@ To get access to the GAE, [see the documentation on Slab](https://uclh.slab.com/
...
-The `pytest.ini` file in the root of the repo contains the configuration for running all tests at once.
-
-We run [pre-commit](https://pre-commit.com/) as part of the GitHub Actions CI. To install and run it locally, do:
-
-```shell
-pip install pre-commit
-pre-commit install
-```
-
-The configuration can be found in [`.pre-commit-config.yml`](./.pre-commit-config.yaml)
```
Comment (p-j-smith) [inline README.md:48]: this has moved to `docs/setup/developer.md` to keep it all in one place. Also clarified that running `pytest` from the root directory doesn't run tests for `hasher`

File: pixl_core/README.md [hunk 3]
```diff
@@ -18,13 +18,13 @@ Specifically, it defines:
...
 ```bash
-pip install -e .
+python -m pip install -e .
 ```
 
 ## Testing
 
 ```bash
-pip install -e .[test] && pip install -e ../pytest-pixl
+pytohn -m pip install -e ../pytest-pixl -e ".[test]"
```
Comment (stefpiatek) [inline pixl_core/README.md:27]:
```suggestion
python -m pip install -e ../pytest-pixl -e ".[test]"
```

File: docs/setup/developer.md (hunk 1 above)
Comment (stefpiatek) [reply docs/setup/developer.md:10]: yeah agreed, its fairly painful because we're using core in pytest pixl

File: README.md (hunk 2 above)
Comment (stefpiatek) [reply README.md:48]: yeah nice, thanks!

File: README.md [hunk 4]
```diff
@@ -15,37 +15,28 @@ To get access to the GAE, [see the documentation on Slab](https://uclh.slab.com/
...
 
 ```shell
-pip install -e cli/
+python -m pip install pixl_core/
+python -m pip install pytest-pixl/
+python -m pip install -e cli/
+python -m pip install pixl_imaging/
+python -m pip install pixl_dcmd/
+python -m pip install pixl_export/
+python -m pip install hasher/
```
Comment (stefpiatek) [inline README.md:27]: So for a deployment you only need to install `core` and `cli`, the others are all used within docker. Then the development instructions can let you know how to install the dev and test

both of these need a `-e`, what's fun is that if you ever install without -e, then it never is installed in editable mode and you need to kill your environment and start again.

**Review by stefpiatek — APPROVED** [review]
Looks good thanks! For non-dev installation only need the CLI installed though

File: README.md (hunk 4 above)
Comment (p-j-smith) [reply README.md]: fun! Just wondering why core also needs to be editable (as far as I could see, only the CLI uses `PIXL_ROOT`)? I've updated the installation instructions now

Comment (stefpiatek) [reply README.md]: ah, that's more of avoiding issues if there's updates to the module that are still in the same version. We've had a lot of fun with that, expecting it to update when it doesn't

Comment (p-j-smith) [reply README.md]: oh I see, that makes sense, thanks

**Review by stefpiatek — APPROVED** [review]
thanks for getting this in

**Pull request merged**